### PR TITLE
Implemented enforcement of code formatting style

### PR DESCRIPTION
**Pull request by silvanocerza**
This should solve #274.
I've not formatted any code in this PR just implemented a way to enforce it, I was thinking of creating a separate PR after this one is merged with a single commit that formats all C++ code.

### CODE REVIEW

File: .github/workflows/format-code.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: "Format C++ code"
```
Comment (patricksjackson) [inline .github/workflows/format-code.yml:1]: What does this workflow do? I _think_ it lets us comment using either of those two format strings on a pull request and have the CI make a formatting commit, is that right?

Comment (silvanocerza) [reply .github/workflows/format-code.yml:1]: Yes, it does exactly that. Something similar is already in use for the C# track and for Markdown and JSON files in the V3 repo.

For example in PR exercism/v3#1788 you can see the one for Markdown and Json in action.

**Comment by silvanocerza** [comment]
No worries about the delay. :)

By the way as soon as I can I'll make another PR formatting the whole repository.

**Comment by patricksjackson** [comment]
Awesome thanks :)

**Pull request merged**